### PR TITLE
(CLOUD-2054) Readme updates.

### PR DESCRIPTION
**Pull request by EamonnTP**
Readme updates include:

- Removed installation section as this is described when published to the Forge.

- The module is now PDK compliant, so I have added the "Validating and unit testing the module" section.

- Updated the Limitations section: updated Ubuntu to 18.04 and added Ruby 2.3.0 or higher.  

- Reorganized parameters and double-checked entries against those in the init.pp. 

- Minor edits elsewhere in the Readme.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -118,6 +96,39 @@ class {'kubernetes':
...
+
+To validate the metadata.json file, run the following command:
+
+```puppet
+pdk validate metadata --puppet-version='5.3.6'
+```
+
+To validate the Puppet code and syntax, run the following command:
+
+```puppet
```
Comment (HelenCampbell) [inline README.md]: These technically aren't puppet code, they are CLI commands. Aside from that the commands themselves and the descriptions look grand :)

File: README.md [hunk 2]
```diff
@@ -7,100 +7,78 @@
...
 
-* `OS`: The operating system Kubernetes runs on.
-* `VERSION`: The version of Kubernetes to deploy.
-* `CONTAINER_RUNTIME`: The container runtime Kubernetes uses. Set this value to `docker` or `cri_containerd`.
-* `CNI_PROVIDER`: The CNI network to install. Set this value to `weave` or `flannel`.
-* `ETCD_INITIAL_CLUSTER`: The server hostnames and IPs in the form of `hostname:ip`. When in production, include three, five, or seven nodes for etcd.
-* `ETCD_IP`: The IP each etcd member listens on. We recommend passing the fact for the interface to be used by the cluster.
-* `KUBE_API_ADVERTISE_ADDRESS`: The IP each etcd/apiserver instance uses on each controller. We recommend passing the fact for the interface to be used by the cluster.
-* `INSTALL_DASHBOARD`: A boolean which specifies whether to install the dashboard.
+* `OS`: The operating system Kubernetes runs on.
```
Comment (HelenCampbell) [inline README.md]: Wee side note, I'd remove all the extra spaces at the end of these sentences, they can trip up the translator so you'd be saving yourself work later on if the module ever got translated.

File: README.md (hunk 1 above)
Comment (EamonnTP) [reply README.md]: Thanks @HelenCampbell

File: README.md (hunk 2 above)
Comment (EamonnTP) [reply README.md]: Thanks @HelenCampbell for pointing this out.

**Pull request merged**